### DISCUSSION TITLE
Running docker setup on Unraid

### DISCUSSION THREAD

**Discussion by MostHated** [supabase/supabase#13786] [Questions]
Hey there, 
    I was just wondering if anyone had attempted to run this on an unraid box? I have the 'compose' plugin, which makes it so you can edit your docker-compose.yml and .env from the ui, but I am having an issue actually firing it up. I have had minimal issues running running other things, but none were quite this complex, so I would verture a guess that there must be something special I need to do, I am just not sure what yet, so figured it was worth an ask if anyone happened to have known.

Naturally, it ran fine on my local machine (pop_os) but unraid is slackware, so there are some times (mostly just app packaging) that custom things need to be done. 

The first time I started it up, it said `/usr/local/emhttp/glcoud.json` was missing, so I created one and tried again, and that error went away, but I don't seem to have that file on my local pc, and didn't remember seeing that error.

currently, there is a repeating log message of:

```log
 19:15:35.268 [info]  Application goth exited: Goth.Application.start(:normal, []) returned an error: shutdown: failed to start child: Goth.Config
    ** (EXIT) an exception was raised:
        ** (Jason.DecodeError) unexpected end of input at position 0
            (jason 1.4.0) lib/jason.ex:92: Jason.decode!/2
            (goth 1.3.0-rc.2) lib/legacy/config.ex:199: Goth.Config.decode_json/1
            (goth 1.3.0-rc.2) lib/legacy/config.ex:66: Goth.Config.load_and_init/1
            (stdlib 3.17.2.2) gen_server.erl:423: :gen_server.init_it/2
            (stdlib 3.17.2.2) gen_server.erl:390: :gen_server.init_it/6
            (stdlib 3.17.2.2) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
{"Kernel pid terminated",application_controller,"{application_start_failure,goth,{{shutdown,{failed_to_start_child,'Elixir.Goth.Config',{#{'__exception__' => true,'__struct__' => 'Elixir.Jason.DecodeError',data => <<>>,position => 0,token => nil},[{'Elixir.Jason','decode!',2,[{file,\"lib/jason.ex\"},{line,92}]},{'Elixir.Goth.Config',decode_json,1,[{file,\"lib/legacy/config.ex\"},{line,199}]},{'Elixir.Goth.Config',load_and_init,1,[{file,\"lib/legacy/config.ex\"},{line,66}]},{gen_server,init_it,2,[{file,\"gen_server.erl\"},{line,423}]},{gen_server,init_it,6,[{file,\"gen_server.erl\"},{line,390}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,226}]}]}}},{'Elixir.Goth.Application',start,[normal,[]]}}}"}
Kernel pid terminated (application_controller) ({application_start_failure,goth,{{shutdown,{failed_to_start_child,'Elixir.Goth.Config',{#{'__exception__' => true,'__struct__' => 'Elixir.Jason.DecodeError',data => <<>>,position => 0,token => nil},[{'Elixir.Jason','decode!',2,[{file,"lib/jason.ex"},{line,92}]},{'Elixir.Goth.Config',decode_json,1,[{file,"lib/legacy/config.ex"},{line,199}]},{'Elixir.Goth.Config',load_and_init,1,[{file,"lib/legacy/config.ex"},{line,66}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,423}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,390}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}}},{'Elixir.Goth.Application',start,[normal,[]]}}})
Crash dump is being written to: erl_crash.dump...done
19:16:03.376 [info]  Starting migration
19:16:04.055 [info]  Application goth exited: Goth.Application.start(:normal, []) returned an error: shutdown: failed to start child: Goth.Config
    ** (EXIT) an exception was raised:   
    ... so on and so fortho
```

That was the main issue I saw, and several of the containers never fire up, so I am guessing that this is holding up the stack from moving forward.
I tried looking up that error, and from what I saw, it came about from misconfiguration somewhere, but not in a file that comes with or is exposed with docker example.
I checked the external ports that are exposed, but the ones that were exposed didn't conflict.

If need be, I could spin up a vm and run it in that, but I would hate to have to run a full fat vm just for one app, but just in case I can't locate a solution, it is an option.

Thanks,
-MH

**Comment by namir-ahmed** [comment]
Hello did you ever figure this out?

**Comment by MostHated** [comment]
Not yet, but I have not tried any further since. Got busy with some other things and had not had a chance to revisit yet.
